pyispyb/core/routes/proposals.py

```diff
@@ -50,92 +50,33 @@
 api_v1.add_namespace(api)
 
 
-__license__ = "LGPLv3+"
-
-@api.route("", endpoint="proposals")
+@api.route("")
 @api.doc(security="apikey")
-class Proposals(Resource):
-    """Allows to get all proposals"""
+@legacy_api.route("/<token>/proposal/list")
```

Comment:
If I may add here a comment, then this is not a good direction... There was numerous discussions about api design and I would suggest to keep it simple. More info here: https://github.com/ispyb/py-ispyb/blob/master/docs/karpics_2020_maxiv_hzb_py-ispyb.pdf and here https://github.com/ispyb/ispyb-web-services/blob/master/Future_Web_Services.md
Also it is a bad idea to pass token in url. Why not keep it as it is in header?

Reply:
Hi @IvarsKarpics 

First, this PR is not to be merged, we would like to organize a meeting (which email I've not sent yet) and we wanted to make public the code we are working on. It is our proposal.

About the API: We agree with you. The only reason why we are proposing doing that is because we want `py-ispyb` to be adopted as soon as possible by the collaboration.

The easiest and quickest way to reach such goal was to be backwards compatible that it means to implement the current ISPyB API into py-ispyb.

In other words, we want people using EXI or EXI2 to switch to py-ispyb out of the box. New functionality like SSX will be built on py-ispyb following the best practices. 

Our rationale was that py-ispyb will take months/years to be finished. Then there are few options:
a) Replace EXI and EXI2 API to use the one from py-ispyb. We will only move to py-ispyb once it is fully finished.
b) Replacing EXI and EXI2 API to use both. As soon as a module, for instance MX, is done in py-ispyb then the EXI/EXI2 API can be replaced.
c) Implement the current API into py-ispyb to be backwards compatible.

Option a) was not ideal as I said because we want people to adopt it asap. Option b) overcomplicates the developments in the UI and it does not warranty adoption.

Then, we took the decision c) because otherwise we don't see a quick and easy adoption (but we might be wrong..). The main drawback of being backward compatible is that we can not modernize the API. Taking into account that ISPyB is in production we thought it was the safest.
As you have noticed they are marked as `legacy` and it will be temporary. Probably, it can be developed as a module that can be dropped at some point. 

Anyway, this is a proposal. We would like to know what people thinks, this is why we are organizing a meeting and why we uploaded the code.